File: src/BaseHmiTypes/Screens/Base/HmiControlWindowBase.cs
namespace BaseHmiTypes.Screens.Base;

public abstract class HmiControlWindowBase : HmiWindowBase
{
    public HmiProperty<HmiColor>? HeaderBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? HeaderForegroundColor { get; set; }

    public HmiProperty<HmiColor>? ContentBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? ContentForegroundColor { get; set; }

    public HmiProperty<HmiHorizontalAlignment>? ContentHorizontalAlignment { get; set; }

    public HmiProperty<HmiVerticalAlignment>? ContentVerticalAlignment { get; set; }

    public HmiFont? HeaderFont { get; set; }

    public HmiFont? ContentFont { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiControlWindowBase.ts
import { HmiColor } from "./HmiColor.js";
import { HmiFont } from "./HmiFont.js";
import { HmiHorizontalAlignment } from "./HmiHorizontalAlignment.js";
import { HmiProperty } from "./HmiProperty.js";
import { HmiVerticalAlignment } from "./HmiVerticalAlignment.js";
import { HmiWindowBase } from "./HmiWindowBase.js";

export abstract class HmiControlWindowBase extends HmiWindowBase {
  headerBackgroundColor?: HmiProperty<HmiColor>;
  headerForegroundColor?: HmiProperty<HmiColor>;
  contentBackgroundColor?: HmiProperty<HmiColor>;
  contentForegroundColor?: HmiProperty<HmiColor>;
  contentHorizontalAlignment?: HmiProperty<HmiHorizontalAlignment>;
  contentVerticalAlignment?: HmiProperty<HmiVerticalAlignment>;
  headerFont?: HmiFont;
  contentFont?: HmiFont;
}
